subworkflows/local/prepare_references.nf

```diff
@@ -43,16 +43,16 @@ workflow PREPARE_REFERENCES {
         BUILD_DICT(ch_fasta)
         ch_dict = BUILD_DICT.out.dict.collect()
 
-        gtf_meta = Channel.fromPath(gtf).map{ it -> [ [id:it[0]], it ] }.collect()
+        gtf_meta = Channel.fromPath(gtf).map{ it -> [ [id:it[0]], it ] }
         GUNZIP_GTF(gtf_meta)
-        ch_gtf_no_meta = gtf.endsWith(".gz") ? GUNZIP_GTF.out.gunzip.map{ meta, gtf -> [gtf] }.collect() : Channel.fromPath(gtf)
+        ch_gtf_no_meta = gtf.endsWith(".gz") ? GUNZIP_GTF.out.gunzip.map{ meta, gtf -> [gtf] }.collect() : Channel.fromPath(gtf).collect()
```

Comment:
```suggestion
        ch_gtf = gtf.endsWith(".gz") ? GUNZIP_GTF.out.gunzip.collect() : ch_gtf
```
Not sure this will work. One would have to change the line above as well

Reply:
I have taken into account and applied in the coming commit